natssysclient/api.go

```diff
@@ -0,0 +1,141 @@
+package natssysclient
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"time"
+
+	"github.com/nats-io/nats.go"
+	"github.com/synadia-io/orbit.go/natsext"
+)
+
+const (
+	// DefaultRequestTimeout is the default timeout for requests, used when the provided context does not have a deadline.
+	DefaultRequestTimeout = 10 * time.Second
+	DefaultStall          = 300 * time.Millisecond
+)
+
+const (
+	srvVarzSubj    = "$SYS.REQ.SERVER.%s.VARZ"
+	srvStatszSubj  = "$SYS.REQ.SERVER.%s.STATSZ"
+	srvConnzSubj   = "$SYS.REQ.SERVER.%s.CONNZ"
+	srvSubszSubj   = "$SYS.REQ.SERVER.%s.SUBSZ"
+	srvHealthzSubj = "$SYS.REQ.SERVER.%s.HEALTHZ"
+	srvJszSubj     = "$SYS.REQ.SERVER.%s.JSZ"
+)
+
+var (
+	ErrValidation      = errors.New("validation error")
+	ErrInvalidServerID = errors.New("sever with given ID does not exist")
+)
+
+// System can be used to request monitoring data from the server.
+type System struct {
+	nc   *nats.Conn
+	opts *sysClientOpts
+}
+
+// SysClientOpt is a functional option for configuring the System client.
+type SysClientOpt func(*sysClientOpts) error
+
+type sysClientOpts struct {
+	multiRequestInterval time.Duration
+	serverCount          int
+}
+
+// StallTimer sets the interval for subsequent responses when pinging the cluster.
+func StallTimer(initialTimeout, interval time.Duration) SysClientOpt {
+	return func(opts *sysClientOpts) error {
+		if interval <= 0 {
+			return fmt.Errorf("%w: interval has to be greater than 0", ErrValidation)
+		}
+		if initialTimeout <= 0 {
+			initialTimeout = interval
+		}
+		opts.multiRequestInterval = interval
+		return nil
+	}
+}
+
+// ServerCount sets the maximum number of servers to wait for the response from.
```

Comment:
I would improve the docs mentioning that this is not necessary most of the time, as the stall should be good enough.

Reply:
Could also mention that the normal PING endpoint includes the overall cluster size as seen by the server. I often just ask my connected server for that and then use that expected size when making many requests in a row